tools: add mosquitto.rsmb MQTT-SN broker

haukepetersen opened the pull request:
### Contribution description
This PR adds the build and a run configuration for the [Mosquitto.rsmb (really small message broker)](https://github.com/eclipse/mosquitto.rsmb), a small MQTT(-SN) gateway and server. Having this in RIOT is very helpful when testing anything MQTT-SN related (as e.g. `examples/emcute` or my upcoming, asynchronous MQTT-SN client implementation). So instead of going through the trouble following the 'setup-guide' in [examples/emcute_mqttsn/README.md](https://github.com/RIOT-OS/RIOT/blob/master/examples/emcute_mqttsn/README.md), one can now simply start a pre-configured MQTT-SN gateway instance by simply issuing
```
make rsmb
```
This will further be useful when designing an automated test setup...

### Issues/PRs references
none yet

miri64 commented on makefiles/tools/targets.inc.mk:
```diff
@@ -4,6 +4,8 @@
 # (as `all`), so `all` will always be the first target defined and thereby the
 # default target when `make` is called without any further argument.
 
+.PHONY: rsmb
```
I have the feeling that this target name is too generic and should maybe be `mqtt_rsmb` or `mosquitto_rsmb` instead.

haukepetersen replied:
`mosquitto_rsmb` it is :-)

haukepetersen commented:
> Some remarks: I can't build from RIOTBASE, only from an application directory. Maybe something to consider?

true, I have never even considered this. Will think about a fix.

haukepetersen commented:
Hmm, maybe not. Making the tool target available from the RIOT root would force us to include the target in the global `Makefile.include` or some related top-level Makefile. This, I think we all agree, is not something we want to do for 'random' deamons, right?! So I'd say we can happily live with the limitation of only being able to trigger the `mosquitto_rsmb` target from within an application, right?!

haukepetersen commented:
renamed target and added logfiles to .gitignore

miri64 commented on dist/tools/mosquitto_rsmb/Makefile:
```diff
@@ -0,0 +1,35 @@
+
+all: git-download
+# Start rsmb build in a clean environment, so variables set by RIOT's build process
+# for cross compiling a specific target platform are reset and edbg can
```
I guess you mean `mosquitto_rsmb` here instead of `edbg`?

miri64 commented on dist/tools/mosquitto_rsmb/Makefile:
```diff
@@ -0,0 +1,35 @@
+all: git-download
+# Start rsmb build in a clean environment, so variables set by RIOT's build process
+# for cross compiling a specific target platform are reset and edbg can
+# be built cleanly for the native platform.
```
Maybe "host platform" instead of "native platform" to distinguish it from the native port?

miri64 commented on dist/tools/mosquitto_rsmb/Makefile:
```diff
@@ -0,0 +1,35 @@
+PKG_NAME     = rsmb
```
Can you change the name here as well please?

miri64 commented:
> Hmm, maybe not. Making the tool target available from the RIOT root would force us to include the target in the global Makefile.include or some related top-level Makefile. This, I think we all agree, is not something we want to do for 'random' deamons, right?! So I'd say we can happily live with the limitation of only being able to trigger the mosquitto_rsmb target from within an application, right?!

Right!

haukepetersen commented:
addressed comments and squashed:
- renamed target and binary to `mosquitto_rsmb`
- fixed comments in Makefile
- remove log files on `make clean`
- copy and remove Message def file now based on wildcard

haukepetersen commented:
tack!
